plugins/GCodeWriter/GCodeWriter.py

```diff
@@ -21,6 +22,40 @@ def write(self, stream, node, mode = MeshWriter.OutputMode.TextMode):
         if gcode_list:
             for gcode in gcode_list:
                 stream.write(gcode)
+            profile = self._serialiseProfile(Application.getInstance().getMachineManager().getActiveProfile()) #Serialise the profile and put them at the end of the file.
+            stream.write(profile)
             return True
 
         return False
+
+    ##  Serialises the profile to prepare it for saving in the g-code.
+    #
+    #   The profile are serialised, and special characters (including newline)
+    #   are escaped.
+    #
+    #   \param profile The profile to serialise.
+    #   \return A serialised string of the profile.
+    def _serialiseProfile(self, profile):
+        version = 1 #IF YOU CHANGE THIS FUNCTION IN A WAY THAT BREAKS REVERSE COMPATIBILITY, INCREMENT THIS VERSION NUMBER!
```

Comment:
See above. But maybe you can use Profile.ProfileVersion instead of using a different version? That would avoid having to keep these two in sync.